src/services/cgnOperatorSearchService.ts

```diff
@@ -0,0 +1,125 @@
+import { OfflineMerchantSearchRequest } from "../../generated/io-cgn-operator-search-api/OfflineMerchantSearchRequest";
+import { OfflineMerchants } from "../../generated/cgn-operator-search/OfflineMerchants";
+export default class CgnService {
+  constructor(
+    private readonly cgnOperatorSearchApiClient: ReturnType<
+      CgnOperatorSearchAPIClient
+    >
+  ) {}
+
+  /**
+   * Get the CGN operator/merchant identified by its identifier.
+   */
+  public readonly getMerchant = (
+    merchantId: NonEmptyString
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<Merchant>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.cgnOperatorSearchApiClient.getMerchant({
+        merchantId
+      });
+
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 404:
+            return ResponseErrorNotFound("Not Found", "Operator Not found");
+          case 500:
+            return ResponseErrorInternal(readableProblem(response.value));
+          default:
+            return ResponseErrorStatusNotDefinedInSpec(response);
+        }
+      });
+    });
+
+  /**
+   * Get an array of CGN online merchants that matches with search criteria
+   * expressed in OnlineMerchantSearchRequest
+   */
+  public readonly getOnlineMerchants = (
+    onlineMerchantSearchRequest: OnlineMerchantSearchRequest
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseSuccessJson<OnlineMerchants>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.cgnOperatorSearchApiClient.getOnlineMerchants(
+        {
+          body: onlineMerchantSearchRequest
+        }
+      );
+
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 500:
+            return ResponseErrorInternal(readableProblem(response.value));
+          default:
+            return ResponseErrorStatusNotDefinedInSpec(response);
+        }
+      });
+    });
+
+  /**
+   * Get an array of CGN offline merchants that matches with search criteria
+   * expressed in OfflineMerchantSearchRequest
+   */
+  public readonly getOfflineMerchants = (
+    offlineMerchantSearchRequest: OfflineMerchantSearchRequest
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseSuccessJson<OfflineMerchants>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.cgnOperatorSearchApiClient.getOfflineMerchants(
+        {
+          body: offlineMerchantSearchRequest
+        }
+      );
+
+      // eslint-disable-next-line sonarjs/no-identical-functions
```

Comment:
Instead of disable `no-identical-functions` we can create a private method inside the service class with the identical code and use it where required.

Reply:
done in https://github.com/pagopa/io-backend/pull/804/commits/7d342ba24b1efeb396af1168d077070de40b0b3e